test/schema-feature.test.js

```diff
@@ -82,6 +82,22 @@ test('The schema should be accessible by id via getSchema', t => {
   fastify.ready(err => t.error(err))
 })
 
+test('The schema should not be undefined in hooks if route does not exist', t => {
+  t.plan(4)
+  const fastify = Fastify()
+
+  fastify.addHook('onRequest', function (req, _reply, done) {
+    t.not(req.routeSchema, undefined)
+    t.not(req.routeOptions.schema, undefined)
```

Comment:
TBH I don't like this default.

How do the user know if a schema has been set?

Now: `if(req.routeSchema==null)`

Then: `if(req.routeSchema.id==null)` + the overhead to create a new empty object


Why don't we fix the types instead?

Reply:
Reopened https://github.com/fastify/fastify/pull/5394, both PRs fix the issue I reported but you guys know best what other use cases to consider.

I go with whatever is decided here :)

Reply:
If we go with this in v4, I’d say right after we should revert this in v5 and merge your other (type) PR there

Reply:
> If we go with this in v4, I’d say right after we should revert this in v5 and merge your other (type) PR there

sounds like a plan, do you want the revert of this to be part of the type PR then? (needs rebase on next first)

Reply:
@gurgunday @Eomm then we go with the other PR. Let's not land and revert.